src/components/MiningPositionList/index.js

```diff
@@ -0,0 +1,263 @@
+          </AutoColumn>
+        </DataText>
+        <DataText area="uniswap">
+          <AutoColumn gap="12px" justify="flex-end">
+            <TYPE.main>{formattedNum(pairPercentage * valueUSD, true, true)}</TYPE.main>
+            <AutoColumn gap="4px" justify="flex-end">
+              <RowFixed>
+                <TYPE.small fontWeight={400}>{formattedNum(pairPercentage * parseFloat(valueFirstPair))} </TYPE.small>
+                <FormattedName
+                  text={firstPairName}
+                  maxCharacters={below740 ? 10 : 18}
+                  margin={true}
+                  fontSize={'11px'}
+                />
+              </RowFixed>
+              <RowFixed>
+                <TYPE.small fontWeight={400}>{formattedNum(pairPercentage * parseFloat(valueSecondPair))} </TYPE.small>
+                <FormattedName
+                  text={secondPairName}
+                  maxCharacters={below740 ? 10 : 18}
+                  margin={true}
+                  fontSize={'11px'}
+                />
+              </RowFixed>
+            </AutoColumn>
+          </AutoColumn>
+        </DataText>
+      </DashGrid>
+    )
+  }
+
+  const miningPositionsSorted =
+    miningPositions &&
+    miningPositions
+
+      .sort((p0, p1) => {
+        if (sortedColumn === SORT_FIELD.VALUE) {
+          const bal0 = (p0.balance / 1e18 / p0.pairData.totalSupply) * p0.pairData.reserveUSD
+          const bal1 = (p0.balance / 1e18 / p0.pairData.totalSupply) * p1.pairData.reserveUSD
+          return bal0 > bal1 ? (sortDirection ? -1 : 1) : sortDirection ? 1 : -1
+        }
+        return 1
+      })
+      .slice(ITEMS_PER_PAGE * (page - 1), page * ITEMS_PER_PAGE)
+      .map((miningPosition, index) => {
+        return (
+          <div key={index}>
+            <ListItem key={index} index={(page - 1) * 10 + index + 1} miningPosition={miningPosition} />
+            <Divider />
+          </div>
+        )
+      })
+
+  return (
+    <ListWrapper>
+      <DashGrid center={true} style={{ height: '32px', padding: 0 }}>
+        {!below740 && (
+          <Flex alignItems="flex-start" justifyContent="flexStart">
+            <TYPE.main area="number">#</TYPE.main>
+          </Flex>
+        )}
+        <Flex alignItems="flex-start" justifyContent="flex-start">
+          <TYPE.main area="number">Name</TYPE.main>{' '}
+        </Flex>
+        <Flex alignItems="center" justifyContent="flexEnd">
+          <ClickableText
+            area="uniswap"
+            onClick={e => {
+              setSortedColumn(SORT_FIELD.VALUE)
+              setSortDirection(sortedColumn !== SORT_FIELD.VALUE ? true : !sortDirection)
+            }}
+          >
+            {below740 ? 'Value' : 'Liquidity'} {sortedColumn === SORT_FIELD.VALUE ? (!sortDirection ? '↑' : '↓') : ''}
+          </ClickableText>
+        </Flex>
+      </DashGrid>
+      <Divider />
+      <List p={0}>{!miningPositionsSorted ? <LocalLoader /> : miningPositionsSorted}</List>
```

Comment:
I think there should also be an empty state added here. If i have no lqiuidity staked it enters perpetual loading. Maybe just some empty card that says something like "No Staked Liquidity" with a link to stake some or something.

Reply:
done